roscpp: "error: 'readv'/'writev' was not declared in this scope" in do_compile

`bitbake roscpp` currently fails on openembedded-core with:
```
 /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot-native/usr/bin/i586-oe-linux/i586-oe-linux-g++  -DROSCONSOLE_BACKEND_LOG4CXX -DROS_PACKAGE_NAME=\"roscpp\" -Droscpp_EXPORTS -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/build/devel/include -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/include -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/build/devel/include/ros -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot/opt/ros/indigo/include -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot/usr/include -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/build  -m32 -march=i586  --sysroot=/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fdebug-prefix-map=/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0=/usr/src/debug/roscpp/1.11.21-r0 -fdebug-prefix-map=/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot-native= -fdebug-prefix-map=/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot=  -fvisibility-inlines-hidden  -m32 -march=i586  --sysroot=/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/recipe-sysroot -I/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/build/devel/include -fPIC   -Wall -Wextra -o CMakeFiles/roscpp.dir/src/libros/transport/transport_udp.cpp.o -c /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp: In member function 'virtual int32_t ros::TransportUDP::read(uint8_t*, uint32_t)':
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp:407:21: error: 'readv' was not declared in this scope
|          num_bytes = readv(sock_, iov, 2);
|                      ^~~~~
[...]
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp: In member function 'virtual int32_t ros::TransportUDP::write(uint8_t*, uint32_t)':
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp:579:25: error: 'writev' was not declared in this scope
|      ssize_t num_bytes = writev(sock_, iov, 2);
|                          ^~~~~~
[...]
| make[2]: *** [CMakeFiles/roscpp.dir/src/libros/transport/transport_udp.cpp.o] Error 1
| make[2]: Leaving directory `/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/build'
| make[1]: *** [CMakeFiles/roscpp.dir/all] Error 2
| make[1]: Leaving directory `/home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/build'
| make: *** [all] Error 2
| WARNING: exit code 2 from a shell command.
| ERROR: Function failed: do_compile (log file is located at /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/temp/log.do_compile.4896)
ERROR: Task (/home/lukas/work/repositories/github.com/bmwcarit/meta-ros/recipes-ros/ros-comm/roscpp_1.11.21.bb:do_compile) failed with exit code '1'
```

The build configuration is:
```
Build Configuration:
BB_VERSION        = "1.34.0"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "ubuntu-14.04"
TARGET_SYS        = "i586-oe-linux"
MACHINE           = "qemux86"
DISTRO            = "nodistro"
DISTRO_VERSION    = "nodistro.0"
TUNE_FEATURES     = "m32 i586"
TARGET_FPU        = ""
meta              = "master:b1c4661742d12cfd444043e597d01bd627ee4ca4"
meta-oe           
meta-multimedia   
meta-python       = "master:66b88f7729d24ef2e3469fcaca172f420c4d54be"
meta-ros          = "master:2b7704917c0443183ea1299b15543157fa0c4185"
```

--- a/recipes-ros/ros-comm/roscpp/0001-roscpp-add-missing-header-for-writev.patch
+++ b/recipes-ros/ros-comm/roscpp/0001-roscpp-add-missing-header-for-writev.patch
@@ -1,0 +1,39 @@
+From 4fbb9f952726a12b1938ea7af9e88ae8b234395e Mon Sep 17 00:00:00 2001
+From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
+Date: Wed, 19 Jul 2017 11:17:11 +0300
+Subject: [PATCH] [roscpp] add missing header for writev().
+
+After an update of gcc and glibc roscpp started to fail builds with the error:
+
+    /home/rojkov/work/ros/build/tmp-glibc/work/i586-oe-linux/roscpp/1.11.21-r0/ros_comm-1.11.21/clients/roscpp/src/libros/transport/transport_udp.cpp:579:25: error: 'writev' was not declared in this scope
+         ssize_t num_bytes = writev(sock_, iov, 2);
+                             ^~~~~~
+
+According to POSIX.1-2001 the function writev() is declared in sys/uio.h.
+
+The patch includes the missing header for POSIX compliant systems.
+
+Upstream-Status: Accepted [https://github.com/ros/ros_comm/pull/1105]
+
+Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
+---
+ clients/roscpp/src/libros/transport/transport_udp.cpp | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/clients/roscpp/src/libros/transport/transport_udp.cpp b/clients/roscpp/src/libros/transport/transport_udp.cpp
+index c7d8298..47d969e 100644
+--- a/clients/roscpp/src/libros/transport/transport_udp.cpp
++++ b/clients/roscpp/src/libros/transport/transport_udp.cpp
+@@ -48,6 +48,9 @@
+ #elif defined(__ANDROID__)
+   // For readv() and writev() on ANDROID
+   #include <sys/uio.h>
++#elif defined(_POSIX_VERSION)
++  // For readv() and writev()
++  #include <sys/uio.h>
+ #endif
+ 
+ namespace ros
+-- 
+2.9.3
+

--- a/recipes-ros/ros-comm/roscpp_1.11.21.bb
+++ b/recipes-ros/ros-comm/roscpp_1.11.21.bb
@@ -10,6 +10,8 @@ DEPENDS = "cpp-common message-generation rosconsole roscpp-serialization roscpp-
 
 require ros-comm.inc
 
+SRC_URI += "file://0001-roscpp-add-missing-header-for-writev.patch;striplevel=3"
+
 ROS_PKG_SUBDIR = "clients"
 
 CXXFLAGS_append = " -I${OECMAKE_BUILDPATH}/devel/include"

